app/multitenant/dynamo_collector.go

```diff
@@ -149,11 +183,30 @@ func NewDynamoDBCollector(
 		bucketName: bucketName,
 		merger:     app.NewSmartMerger(),
 		cache:      gcache.New(reportCacheSize).LRU().Expiration(reportCacheExpiration).Build(),
+		memcache:   memcacheClient,
 		nats:       nc,
 		waiters:    map[watchKey]*nats.Subscription{},
 	}, nil
 }
 
+// updateMemcacheServers sets a memcache server list from SRV records. SRV
+// priority & weight are ignored.
+func updateMemcacheServers(hostname, service string, serverList *memcache.ServerList) error {
+	_, addrs, err := net.LookupSRV(service, "tcp", hostname)
+	if err != nil {
+		return err
+	}
+	var servers []string
+	for _, srv := range addrs {
+		servers = append(servers, fmt.Sprintf("%s:%d", srv.Target, srv.Port))
+	}
+	// ServerList deterministically maps keys to _index_ of the server list.
+	// Since DNS returns records in different order each time, we sort to
+	// guarantee best possible match between nodes.
+	sort.Strings(servers)
+	return serverList.SetServers(servers...)
+}
```

Comment:
I'd like to encapsulate this behaviour in a separate file / type please, much like we do with the consul client.
 Add a line note